src/debug/providerRegistry.ts

```diff
@@ -27,14 +30,38 @@ async function showProviderPick(): Promise<IDebugProvider | undefined> {
     return pickedProvider.provider;
 }
 
-export async function getDebugProvider(baseImage?: string): Promise<IDebugProvider | undefined> {
+export async function getDebugProvider(baseImage?: string, runningProcesses?: ProcessInfo[]): Promise<IDebugProvider | undefined> {
     let debugProvider = null;
     if (baseImage) {
         debugProvider = supportedProviders.find((provider) => provider.isSupportedImage(baseImage));
     }
-    if (!debugProvider) {
-        debugProvider = await showProviderPick();
+
+    if (!debugProvider)
+    {
+        let candidateProviders: IDebugProvider[];

```

Comment:
we are filtering the `debugproviders` shown in picker, based on the processes running on the container if we know them.  If `filterSupportedProcesses` returns `undefined` then it will always show that `debugprovider`